Crypto/PublicKey+Apple.mm

```diff
@@ -392,12 +392,26 @@ virtual int _sign(int/*mbedtls_md_type_t*/ mbedDigestAlgorithm,
                 if (!privateKeyRef)
                     return nullptr;
                 
+                ++gC4ExpectExceptions;
                 auto publicKeyRef = SecKeyCopyPublicKey(privateKeyRef);
+                --gC4ExpectExceptions;
                 if (!publicKeyRef) {
-                    CFRelease(privateKeyRef);
-                    throwMbedTLSError(MBEDTLS_ERR_X509_INVALID_FORMAT); // Impossible?
+                    // If the public key is not in the KeyChain, SecKeyCopyPublicKey() will return null.
+                    // Create a SecKeyRef directly from the publicKey data instead:
+                    NSDictionary* attrs = @{
+                        (id)kSecAttrKeyType:            (id)kSecAttrKeyTypeRSA,
+                        (id)kSecAttrKeyClass:           (id)kSecAttrKeyClassPublic,
+                        (id)kSecAttrKeySizeInBits:      @2048
```

Comment:
This shouldn't be here, should it? The key size isn't fixed, it depends on the actual key data.